plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java

```diff
@@ -233,93 +330,105 @@ public void setEnabled(boolean enabled) {
     stagingLocationInput.setEnabled(enabled);
   }
 
+
+  /**
+   * Return the fetched staging locations result future, or null if no job is yet scheduled.
+   */
+  private synchronized Future<SortedSet<String>> getStagingLocationsResult() {
+    return fetchStagingLocationsJob == null ? null : fetchStagingLocationsJob.getStagingLocations();
+  }
+
   /**
    * Fetch the staging locations from GCS in a background task and update the Staging Locations
    * combo.
    */
-  private void updateStagingLocations(String project) {
+  @VisibleForTesting
+  synchronized void updateStagingLocations(String project, long scheduleDelay) {
```

Comment:
I think this method is called exclusively by the UI thread. My preference is to not use `synchronized` if it is not needed. It makes people wonder the keyword is there for a good reason. People don't use `synchronized` for SWT listener methods. I believe the main reason for the single UI thread architecture is to avoid complex concurrency management such as using `synchronized`.